capsules/src/fm25cl.rs

```diff
@@ -48,8 +49,8 @@ pub trait FM25CLClient {
     fn done(&self, buffer: &'static mut [u8]);
 }
 
-pub struct FM25CL<'a> {
-    spi: &'a hil::spi::SPIMasterDevice,
+pub struct FM25CL<'a, S: hil::spi::SpiMasterDevice + 'a> {
+    spi: &'a S,
```

Comment:
This is totally fine since it saves us a virtual-table, but just to be clear, it's not a necessary change (and you had advocated against this style at some point on account of it being overly verbose). I'm completely fine with it, so up to you for this driver.